### DISCUSSION TITLE
Followed exact instruction, web interface unreachable

### DISCUSSION THREAD

**Discussion by odoo-mvds** [postalserver/postal#3046] [Help with using Postal]
Hi,

Someone an idea? I followed the exact instruction of the website step by step.
I see caddy running and listening on port 443. My DNS is setup correctly, but i'm not getting the web interface.
Any tip is appreciated!
<img width="571" alt="image" src="https://github.com/user-attachments/assets/b2f2f85f-ceb9-4fdd-89d8-ff4574fd9634">

BR

**Comment by fornof** [comment]
I ran into this issue, I found the host matters
```
 curl -vvv -H 'host:YOURHOSTNAME_FROM_CONFIGS' http://localhost:5000/login?return_to=%2F , gives html
```
hope that helps a little bit. 
I'm using 443 for kubernetes, so this may need to get an endpoint forwarded, but just 5000 seems to go somewhere to start. 

```html
*   Trying 127.0.0.1:5000...
* Connected to localhost (127.0.0.1) port 5000 (#0)
> GET /login?return_to=%2F HTTP/1.1
> Host:mail.vertexmatrix.com
> User-Agent: curl/7.74.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Link: </assets/application/application-d582f69de04a4a1ddabf46f496aea06e97cee7e2d7e29b856091c57dfc9ec98d.css>; rel=preload; as=style; nopush,</assets/application/application-0dd293ea0a8f7f37880ae5bea4a4f1627195bd86b17430b7fb0313790f195df8.js>; rel=preload; as=script; nopush
< Content-Type: text/html; charset=utf-8
< Vary: Accept
< ETag: W/"63310a9995505663ff511376d3876376"
< Cache-Control: max-age=0, private, must-revalidate
< Set-Cookie: browser_id=9bb5c428-ab75-4b88-b7d3-e3c4f9a05f13; path=/; expires=Mon, 16 Jul 2029 22:32:03 GMT; HttpOnly; SameSite=Lax
< Set-Cookie: _postal_session=WT4tKSoBFSBcLVk0gwkag91W%2FH4q8leCvyXio9v85IJb2JqpqMvAd2y8rGiwd1M2MLvSGsL0YogYgHEQghhlSmrdOVSH98QR%2Fe5SArtGtFzUXv3jX70QxKjH7FDbU5GhzSiIVh6wfLz6W62lIXsuQ1q3lbziqlCAQONcpYuPWGb3z%2BCz%2FafYXrGw4vh0Tw2%2Fb2yOn2num2Afbmi2Ulv5ii4SJ1eM5CgeGFzKEzdELxYN%2FO%2Bad8AmYzPa53tMPwUmXcp4Lkf4km9IO%2FRVyvH4XG9vd%2B0T77w%3D--%2BjB%2BGmI8tMS6ZswR--NTq5iJ%2BAEt3tNlxfkGEQKA%3D%3D; path=/; HttpOnly; SameSite=Lax
< X-Request-Id: 5eed7808-e4c9-4842-b07e-32bcae94fdee
< X-Runtime: 0.014241
< X-Frame-Options: sameorigin
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< X-Download-Options: noopen
< X-Permitted-Cross-Domain-Policies: none
< Content-Security-Policy: child-src 'self'; connect-src 'self'; font-src 'self' https: data:; img-src 'self' https: data:; object-src 'none'; script-src 'self'; style-src 'self' https: 'unsafe-inline'
< Content-Length: 1940
<
<!DOCTYPE html>
<html class='subPage'>
<head>
<title>Login - Postal</title>
<meta name="csrf-param" content="authenticity_token" />
<meta name="csrf-token" content="IaJzWmN3jGdxnJ2kh8xiBaXLb-l_m1w_JJE_-ACFFxPbtg_JFuRMSJbbBM5HO9TRxBe1-jX_ZMRHXBhHxIadTw" />
<link rel="stylesheet" href="/assets/application/application-d582f69de04a4a1ddabf46f496aea06e97cee7e2d7e29b856091c57dfc9ec98d.css" data-turbolinks-track="reload" />
<script src="/assets/application/application-0dd293ea0a8f7f37880ae5bea4a4f1627195bd86b17430b7fb0313790f195df8.js" data-turbolinks-track="reload"></script>
<link href='/assets/favicon-cbf06fae3095e2043b79ae5eaa186df6c4dbaa0a2e5d6fef88baae67cc78fd79.png' rel='shortcut icon'>
<meta name="turbolinks-cache-control" content="no-cache">
</head>
<body>
<div class='subPageBox'>
<div class='subPageBox__title'>
Welcome to Postal
</div>

<div class='subPageBox__content'>
<form class="loginForm" action="/login" accept-charset="UTF-8" method="post"><input type="hidden" name="authenticity_token" value="IETbqu9wPUVZqjw54xzp_FtcTpdRrs0mRwHY5bw8iO3IiU0zw7oLHKf8oW2Hjzl8T3fqm_l2fXOEMf5_WF0-Dg" autocomplete="off" /><input type="hidden" name="return_to" id="return_to" value="/" autocomplete="off" />
<p class='loginForm__input'><input type="email" name="email_address" id="email_address" value="" spellcheck="false" class="input input--text input--onWhite" placeholder="Your e-mail address" autofocus="autofocus" tabindex="1" /></p>
<p class='loginForm__input'><input type="password" name="password" id="password" value="" class="input input--text input--onWhite" placeholder="Your password" tabindex="2" /></p>
<div class='loginForm__submit'>
<ul class='loginForm__links'>
<li><a href="/login/reset?return_to=%2F">Forgotten your password?</a></li>
</ul>
<p><input type="submit" name="commit" value="Login" class="button button--positive" tabindex="3" data-disable-with="Login" /></p>
</div>
</form></div>

</div>
</body>
</html>
* Connection #0 to host localhost left intact
```

**Comment by willpower232** [comment]
When you say you are not getting the web interface, what do you see in your browser?

**Comment by odoo-mvds** [comment]
Hi,

@fornof  running your command worked and returned me the same result. So the page is locally available. However I can't connect to it remotely... No firewall running, and it's a clean install on debian vultr VPS...

@willpower232 the page is loading indefinitely. You can see in the chrome bar the circle turning but no page showing up. I also tried http:// on port 5000 but also no luck there...

br,

**Reply by willpower232** [reply]
Port 5000 should not be public so that won't help you. Have you definitely got the DNS records pointing at your server?

**Reply by odoo-mvds** [reply]
Hi,

Yes 100% sure the DNS is fine. I can connect to my server using SSH and that DNS. Seems the issue is somewhere caddy related, as locally i can get the html file using the command of @fornof.

I'm trying the same command of @fornof but with the https enabled and i'm getting this:

root@mail:/opt/postal/config# curl -vvv -H 'host:MY HOSTNAME' https://localhost:443/login?return_to=%2F
*   Trying 127.0.0.1:443...
* Connected to localhost (127.0.0.1) port 443 (#0)
* ALPN: offers h2,http/1.1
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
*  CAfile: /etc/ssl/certs/ca-certificates.crt
*  CApath: /etc/ssl/certs
* TLSv1.3 (IN), TLS alert, internal error (592):
* OpenSSL/3.0.13: error:0A000438:SSL routines::tlsv1 alert internal error
* Closing connection 0
curl: (35) OpenSSL/3.0.13: error:0A000438:SSL routines::tlsv1 alert internal error

Any idea?

br,

**Reply by fornof** [reply]
what happens when you try that with the -k (ignore tls) flag? 
curl -v -H 'host:MY HOSTNAME' https://127.0.0.1:443 -k

**Reply by odoo-mvds** [reply]
*   Trying 127.0.0.1:443...
* connect to 127.0.0.1 port 443 failed: Connection refused
* Failed to connect to 127.0.0.1 port 443 after 0 ms: Couldn't connect to server
* Closing connection 0
curl: (7) Failed to connect to 127.0.0.1 port 443 after 0 ms: Couldn't connect to server

**Reply by odoo-mvds** [reply]
sorry, Caddy wasn't running: got the same error.
*   Trying 127.0.0.1:443...
* Connected to 127.0.0.1 (127.0.0.1) port 443 (#0)
* ALPN: offers h2,http/1.1
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* TLSv1.3 (IN), TLS alert, internal error (592):
* OpenSSL/3.0.13: error:0A000438:SSL routines::tlsv1 alert internal error
* Closing connection 0
curl: (35) OpenSSL/3.0.13: error:0A000438:SSL routines::tlsv1 alert internal error

**Reply by fornof** [reply]
well, with your command and the k flag
```
curl -vvv -k -H 'host:MY HOSTNAME' https://localhost:443/login?return_to=%2F
```
can you make sure -k is in your command? it skips tls verification

**Reply by odoo-mvds** [reply]
when i try sudo apt update i get some certificate errors:

W: Failed to fetch https://deb.debian.org/debian/dists/bookworm/InRelease  Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown.  Could not handshake: Error in the certificate verification. [IP: 146.75.122.132 443]
W: Failed to fetch https://deb.debian.org/debian-security/dists/bookworm-security/InRelease  Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown.  Could not handshake: Error in the certificate verification. [IP: 146.75.122.132 443]
W: Failed to fetch https://deb.debian.org/debian/dists/bookworm-updates/InRelease  Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown.  Could not handshake: Error in the certificate verification. [IP: 146.75.122.132 443]
W: Failed to fetch https://debian.mirror.constant.com/dists/bookworm/InRelease  Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown.  Could not handshake: Error in the certificate verification. [IP: 108.61.5.92 443]
W: Failed to fetch https://download.docker.com/linux/debian/dists/bookworm/InRelease  Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown.  Could not handshake: Error in the certificate verification. [IP: 18.239.94.125 443]
W: Failed to fetch https://apprepo.vultr.com/debian/dists/universal/InRelease  Certificate verification failed: The certificate is NOT trusted. The certificate issuer is unknown.  Could not handshake: Error in the certificate verification. [IP: 45.63.19.147 443]
W: Some index files failed to download. They have been ignored, or old ones used instead.

So maybe the debian image of vultre has issues. I think i will try to reinstall it with another version of linux, Ubuntu.

BR,

**Reply by odoo-mvds** [reply]
even with the -k command the error remains the same.

I will try a clean install of the machine. Which Linux distro do u suggest to use?

br,

**Reply by fornof** [reply]
you could also try updating caddy : https://caddy.community/t/internal-ssl-error-instead-of-warning-for-invalid-cert/23872/5

**Reply by fornof** [reply]
you can also look at this answer to see if it helps: https://askubuntu.com/questions/1095266/apt-get-update-failed-because-certificate-verification-failed-because-handshake
